### PR TITLE
Add preview feedback and cleanup

### PR DESCRIPTION
**Pull request by ellie**
1) React to deploy request comment
2) Comment with deploy URL once complete
3) Cleanup deploy on comment 

Auto cleanup on merge to follow :)

## Type of change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/workflows/pr-deploy.yaml [hunk 1]
```diff
@@ -63,3 +70,30 @@ jobs:
...
+        if: ${{ steps.check.outputs.has-permission && github.event.comment.body == '!deploy'}} 
+
+      # Only ran if the correct comment is used! We should have automatic cleanup on merge, but this allows us to manually
+      # cleanup. For instance, you may restore a branch, deploy it, and keep it around for demo purposes.
+      - name: Delete preview namespace
+        id: delete_preview
+        run: |
+          doctl kubernetes cluster kubeconfig save $PR_CLUSTER_ID 
+
+          helm uninstall -n pr-${{ steps.vars.outputs.git_short }} posthog
```
Comment (guidoiaquinti) [inline .github/workflows/pr-deploy.yaml:90]: I'll be happy if this works https://github.com/PostHog/charts-clickhouse/issues/305. For Kubetest we currently have a workaround to patch some k8s definition

File: .github/workflows/pr-deploy.yaml [hunk 2]
```diff
@@ -63,3 +70,30 @@ jobs:
...
+        uses: peter-evans/create-or-update-comment@c9fcb64660bc90ec1cc535646af190c992007c32
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            Change deployed and available at https://${{ steps.vars.outputs.fqdn_record }}
+
+            Comment with `!deploy-cleanup` to delete your deployment!
+        if: ${{ steps.check.outputs.has-permission && github.event.comment.body == '!deploy'}} 
+
+      # Only ran if the correct comment is used! We should have automatic cleanup on merge, but this allows us to manually
```
Comment (hazzadous) [inline .github/workflows/pr-deploy.yaml:83]: If you need examples of destroy on merge there is something in the posthog-cloud-infra on merge of PRs to clean up test resources.

**Review by hazzadous — APPROVED** [review]
So you can test this on PR's perhaps we can add a trigger on pull request label change to `deploy` or something.

**Comment by ellie** [comment]
> So you can test this on PR's perhaps we can add a trigger on pull request label change to `deploy` or something.

I'm pretty sure that would have the same effect, we'd need it to be on push or similar :/ All normal events (like the kind you'd otherwise use webhooks to handle) seem to just run whatever is on main

**Comment by ellie** [comment]
!deploy

**Comment by hazzadous** [comment]
> I'm pretty sure that would have the same effect, we'd need it to be on push or similar :/ All normal events (like the kind you'd otherwise use webhooks to handle) seem to just run whatever is on main

@ellie `labeled` is an activity type of the `pull_request` event and as such will run on the branch afaict https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

**Comment by ellie** [comment]
@hazzadous ahhh ok I'll give that one a try! ty 🙌 

https://18e550d.pr.posthog.cc/ is live :)

**Comment by ellie** [comment]
!deploy-cleanup

**Pull request merged**